src/main/java/com/google/sps/DataGraph.java

```diff
@@ -107,9 +117,14 @@ public DataGraph getCopy() {
     for (String key : graphNodesMap.keySet()) {
       copyMap.put(key, graphNodesMap.get(key));
     }
+
+    HashMap<String, Set<String>> tokenCopyMap = new HashMap<>();
+    for (String key : this.tokenMap().keySet()) {
+      tokenCopyMap.put(key, this.tokenMap().get(key));
```

Comment:
Since we want our internal data structures to be "immutable" for a data graph, when we create a copy of this map should we also create a copy of the set for each token. I know that in add/remove token we modify this set, which would result in earlier versions possibly being modified as well.